pkg/core/resources/apis/meshmultizoneservice/api/v1alpha1/validator.go

```diff
@@ -13,5 +16,13 @@ func (r *MeshMultiZoneServiceResource) validate() error {
 	if len(r.Spec.Ports) == 0 {
 		verr.AddViolationAt(path.Field("ports"), "cannot be empty")
 	}
+	for i, port := range r.Spec.Ports {
+		if port.Name != nil && *port.Name == "" {
+			verr.AddViolationAt(path.Field("ports").Index(i).Field("name"), validators.MustNotBeEmpty)
+		}
+		if !slices.Contains(mesh.SupportedProtocols, port.AppProtocol) {
```

Comment:
it's a bit concerning MES allows `tcp;grpc;http;http2` but `mesh.SupportedProtocols` also has `kafka`

Reply:
https://github.com/kumahq/kuma/issues/12912 added issue about this